src/timeseriesUpdate.ts

```diff
@@ -73,3 +89,60 @@ export function updateNTokenPresentValueHistoricalData(notional: Notional, curre
   log.debug('Updated nTokenPresentValueHistoricalData variables for entity {}', [nTokenPresentValueHistoricalData.id]);
   nTokenPresentValueHistoricalData.save();
 }
+
+export function updateTvlHistoricalData(notional: Notional, maxCurrencyId: i32, timestamp: i32): void {
+  let tvlCurrencies = new Array<string>();
+  let usdTotal = BigInt.fromI32(0);
+
+  for (let currencyId: i32 = 1; currencyId <= maxCurrencyId; currencyId++) {
+    let result = notional.try_getCurrencyAndRates(currencyId);
+    if (!result.reverted) {
+      let assetToken = result.value.value0;
+      let erc20 = ERC20.bind(assetToken.tokenAddress);
+      let assetTokenBalance = erc20.balanceOf(notional._address);
+      let underlyingValue = convertAssetToUnderlying(notional, currencyId, assetTokenBalance);
+      let ethRate = result.value.value2;
+      let ethRateDecimals = result.value.value2.rateDecimals;
+      let ethValue = underlyingValue.times(ethRate.rate).div(ethRateDecimals).div(assetToken.decimals);
```

Comment:
I don't believe the second `div` is required. It should just be:
`let ethValue = underlyingValue.times(ethRate.rate).div(ethRateDecimals)`

Reply:
If I remove the second division, do I need to add it to the eth to usd conversion?

Reply:
I don't believe it is necessary in either, does it cause any issues if its removed?

Reply:
No issues, I was having a hard time validating values with kovan. I'll fix this.